builtin/logical/database/path_creds_create.go

```diff
@@ -107,21 +101,75 @@ func (b *databaseBackend) pathCredsCreateRead() framework.OperationFunc {
 			RollbackStatements: v5.Statements{
 				Commands: role.Statements.Rollback,
 			},
-			Password:   password,
 			Expiration: expiration,
 		}
 
-		// Overwriting the password in the event this is a legacy database plugin and the provided password is ignored
+		respData := make(map[string]interface{})
+
+		// Generate the credential based on the role's credential type
+		switch role.CredentialType {
+		case v5.CredentialTypePassword:
```

Comment:
I think that we need to also include `""` for existing roles that didn't have this set prior to this field existing.

Reply:
[CredentialType](https://github.com/hashicorp/vault/blob/db-secrets-credential-types/sdk/database/dbplugin/v5/database.go#L124-L130) is intentionally an `int` so that we don't have to check for the empty string on existing roles. The zero value will default to `CredentialTypePassword` for existing roles, which is what we want.